### PR TITLE
Add 'init' option for docker_swarm_service.

### PR DESCRIPTION
**Pull request by pklejch**
##### SUMMARY
Added support for init option, which corresponds to --init option in docker service create.
```
docker_swarm_service:
  init: true
```
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
docker_swarm_service

### CODE REVIEW

File: plugins/modules/cloud/docker/docker_swarm_service.py [hunk 1]
```diff
@@ -187,6 +187,13 @@
       - Service image path and tag.
       - Corresponds to the C(IMAGE) parameter of C(docker service create).
     type: str
+  init:
+    description:
+      - Use an init inside each service container to forward signals and reap processes.
+      - Corresponds to the C(--init) option of C(docker service create).
+      - Requires API version >= 1.37.
+    type: bool
+    version_added: "2.10"
```
Comment (felixfontein) [inline plugins/modules/cloud/docker/docker_swarm_service.py:196]:
```suggestion
```
Please remove this line. We're adding `version_added` later once the next value is determined.

File: plugins/modules/cloud/docker/docker_swarm_service.py [hunk 2]
```diff
@@ -2434,6 +2448,8 @@ def get_service(self, name):
                 ds.networks.append(network)
         ds.service_version = raw_data['Version']['Index']
         ds.service_id = raw_data['ID']
+
+        ds.init = task_template_data['ContainerSpec'].get('Init')
```
Comment (felixfontein) [inline plugins/modules/cloud/docker/docker_swarm_service.py:2452]: I'm not sure whether this makes sense, but maybe:
```suggestion
        ds.init = task_template_data['ContainerSpec'].get('Init', False)
```

Comment (pklejch) [reply plugins/modules/cloud/docker/docker_swarm_service.py]: Yes, I agree your suggestion makes more sense.

**Comment by felixfontein** [comment]
@pklejch thank for implementing this, and moving the PR!
@hannseman thanks for reviewing!

**Pull request merged**